### PR TITLE
Feature/remove pid

### LINKED ISSUE #51: Remove one of the PID controllers
we currently have two classes in control which are PID's. Remove one, keep the other, and use that one everywhere.

### PR DESCRIPTION
**Pull request by ThijsLuttikhuis**
### Summary
-> remove "PID" from control because there were two of them.
-> fix PID values using the new "Controller"
-> improve Attack Skill

fixes #51 

### Comments for the reviewer
There is a chance that I might have the idea that I could potentially slightly dislike Pull Request

### Checklist for the reviewer
This acts as a guide to help you and/or the reviewer to analyze the code. 

**functionality**  
- [ ] The code is properly tested
- [ ] The tests are passing
- [ ] The code works as expected
- [ ] The code is understandable and easy to read
- [ ] Proper abstractions
- [ ] No duplicate code

**style**  
- [ ] Variables/objects/classes have proper names
- [ ] No unused variables
- [ ] No magic numbers
- [ ] The code is well documented
- [ ] No unused imports
- [ ] No leftover code / unused comments
- [ ] Proper indentation
- [ ] Proper use of namespaces

### CODE REVIEW

File: roboteam_ai/src/skills/Attack.cpp [hunk 1]
```diff
@@ -23,17 +23,25 @@ bt::Node::Status Attack::onUpdate() {
...
         targetPos = behindBall;
+        goToPos.goToPos(robot, targetPos, GoToType::luTh);
     } else if (!Coach::doesRobotHaveBall(robot->id, true)) {
         targetPos = ball;
+        goToPos.goToPos(robot, targetPos, GoToType::basic);
     } else {
         targetPos = ball;
         unsigned char forced_kick = 1;
         kicker.kick(robot, forced_kick);
+        goToPos.goToPos(robot, targetPos, GoToType::basic);
```
Comment (mrlukasbos) [inline roboteam_ai/src/skills/Attack.cpp]: Why is this added?

File: roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp [hunk 2]
```diff
@@ -111,12 +111,11 @@ ControlGoToPosLuTh::Command ControlGoToPosLuTh::goToPos(RobotPtr robot, Vector2
...
         if (! pidInit) {
             pidInit = true;
-            pid.initialize(1.0/rtt::ai::constants::tickRate);
-            pid.setParams(3.0, 0.0, 0.2, 0.0, 0.0, 0.0);
+            pid.setPID(3.0, 0, 0.5);
         }
 
         Vector2 pidPos = me.posData[toStep];
-        Vector2 vel = pid.posControl(robot->pos, pidPos, robot->vel);
+        Vector2 vel = pid.controlPID2(static_cast<Vector2>(target - robot->pos));
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp]: `controlPID2` Is een vreemde naam

File: roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp [hunk 3]
```diff
@@ -111,12 +111,11 @@ ControlGoToPosLuTh::Command ControlGoToPosLuTh::goToPos(RobotPtr robot, Vector2
 
         if (! pidInit) {
             pidInit = true;
-            pid.initialize(1.0/rtt::ai::constants::tickRate);
-            pid.setParams(3.0, 0.0, 0.2, 0.0, 0.0, 0.0);
+            pid.setPID(3.0, 0, 0.5);
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp]: Why these values?

File: roboteam_ai/src/skills/Attack.cpp (hunk 1 above)
Comment (ThijsLuttikhuis) [reply roboteam_ai/src/skills/Attack.cpp]: basic gebruik ik voor low level control, luth voor high level / grote afstand

File: roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp (hunk 2 above)
Comment (ThijsLuttikhuis) [reply roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp]: vraag maar aan koen. ik heb de oude alleen verwijderd.

File: roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp (hunk 3 above)
Comment (ThijsLuttikhuis) [reply roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp]: yes. - pid is gewoon magic numbers. niks aan te doen

Comment (mrlukasbos) [reply roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp]: They still can be structured, i.e. by putting them on top of the file as constants or in the header file.

File: roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp (hunk 2 above)
Comment (mrlukasbos) [reply roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp]: Can you refactor it maybe to have a better name?

File: roboteam_ai/src/skills/Attack.cpp (hunk 1 above)
Comment (mrlukasbos) [reply roboteam_ai/src/skills/Attack.cpp]: I mean the lines below it. looks like you added extra functionality.

Comment (mrlukasbos) [reply roboteam_ai/src/skills/Attack.cpp]: Those additions then don't belong in this PR. This PR is about removing a PID controller. (might you be asking why: later on this PR will be converted to one big commit on development, thus making it difficult to trace an issue if it would be there)

File: roboteam_ai/src/control/ControlGoToPos.cpp [hunk 4]
```diff
@@ -1,3 +1,5 @@
+#include <utility>
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/ControlGoToPos.cpp]: include in cpp file

File: roboteam_ai/src/control/ControlGoToPos.cpp [hunk 5]
```diff
@@ -116,12 +118,9 @@ void ControlGoToPos::goToPosForce(RobotPtr robot, Vector2 &targetPos) {
 }
 
 void ControlGoToPos::goToPosLuTh(RobotPtr robot, Vector2 &targetPos) {
-    if ((static_cast<Vector2>(robot->pos) - targetPos).length() < 0.50) {
-        goToPosBasic(robot, targetPos);
-    } else {
-        Command command = gtpLuth.goToPos(robot, targetPos);
-        publishRobotCommand(command);
-    }
+
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/ControlGoToPos.cpp]: Why did this change?

File: roboteam_ai/src/control/Controller.h [hunk 6]
```diff
@@ -55,14 +55,14 @@ namespace rtt{
...
+                double controlPID(double err, double rate);
 
                 //Same as above but for when we want to control 2 variables with the same k values
                 //basically x and y
-                Vector2 controlPID2(Vector2 err);
+                Vector2 controlPID(Vector2 err);
 
                 //To fill in your own measured velocity
-                Vector2 controlPIR2(Vector2 err, Vector2 rate);
+                Vector2 controlPID(Vector2 err, Vector2 rate);
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/Controller.h]: The 'D' needs to change to an 'R' again...

File: roboteam_ai/src/skills/Attack.cpp [hunk 7]
```diff
@@ -22,27 +22,29 @@ bt::Node::Status Attack::onUpdate() {
...
+    else {
         roboteam_msgs::RobotCommand command;
         command.id = robot->id;
         command.use_angle = 1;
-        command.w = static_cast<float>((ball-behindBall).angle());
+        command.w = static_cast<float>((ball - behindBall).angle());
         publishRobotCommand(command);
-    }
+        targetPos = ball;
+        goToPos.goToPos(robot, targetPos, GoToType::basic);
```
Comment (mrlukasbos) [inline roboteam_ai/src/skills/Attack.cpp]: are these functionality changes?

File: roboteam_ai/src/control/ControlGoToPos.cpp (hunk 4 above)
Comment (kJh19) [reply roboteam_ai/src/control/ControlGoToPos.cpp]: moved it to h file

File: roboteam_ai/src/skills/Attack.cpp (hunk 7 above)
Comment (kJh19) [reply roboteam_ai/src/skills/Attack.cpp]: No, now two different commands will be sent to the robot after each other, this should be changed.
